feat: Allow users to override secure boot key and password

noelmiller opened the pull request:
# Purpose

This PR is allowing a user to override the secure boot key and secure boot password to allow them to enroll their own key.

Fixes: #32 

# What I did
- Updated tests to include a test for secure-boot ISOs
- Removed unneeded hardcoded files
- Added Variables for SECURE_BOOT_KEY_URL and ENROLLMENT_PASSWORD
- Added checks to pull down secure boot key from SECURE_BOOT_KEY_URL

p5 commented:
Nice PR!

In order for us to benefit from these changes, we'll need to make some other changes to the action to remove hard-coded refs, but this is a great start!

https://github.com/ublue-os/isogenerator/blob/main/action.yml#L66

akdev1l commented:
thank you for pointing that out! I will fix in a follow up :)

noelmiller committed:
feat: Allow users to override secure boot key and password when using container or action

noelmiller committed:
fix: follow redirects

noelmiller committed:
chore: update docs

noelmiller committed:
fix: added comments and removed additional \n

noelmiller committed:
fix: removed defaults

noelmiller committed:
fix: added conditional for adding public key

noelmiller committed:
chore(ci): Added additional test for secure boot

noelmiller committed:
chore(ci): fixed up workflows to match production

noelmiller committed:
fix(ci): added workflow dispatch to test

noelmiller committed:
fix(ci): fixed version and added IMAGE_TAG

noelmiller committed:
chore(ci): changed to bluefin to test both 38 and 39 secure boot

noelmiller committed:
Merge branch 'main' into decouple_secure_boot_key

noelmiller committed:
chore(ci): added required variables to entrypoint

noelmiller committed:
chore(ci): added merge_group and added paths-ignore

noelmiller committed:
chore(ci): updated description

noelmiller committed:
chore(ci): set to Silverblue since we are testing Bluefin

noelmiller commented:
crap, not ready for review. disregard.

noelmiller committed:
chore(ci): Fixed secure boot key not found error

Received error in anaconda when check was hit, need to set as a soft failure to exit the script early without stopping anaconda installation.

noelmiller committed:
chore: fixed formatting

noelmiller committed:
chore: fixed whitespace

noelmiller committed:
chore(ci): Removed duplicate test

noelmiller committed:
chore: fix whitespace

noelmiller committed:
chore(ci): fixed test and removed upload to Github Artifacts

noelmiller committed:
chore: updated README

noelmiller commented:
OK ready for review. For now I am going to stick with a URL since the primary purpose of enrolling a key is to allow our kmods to work. Most users will not require this level of customization and would use our key anyway if they were building off any of our base images with our kmods.

bsherman commented on Makefile:
```diff
@@ -8,6 +8,8 @@ IMAGE_TAG = $(VERSION)
 EXTRA_BOOT_PARAMS =
 VARIANT = Kinoite
 WEB_UI = false
+ENROLLMENT_PASSWORD = isogenerator
```
I know I suggested "isogenerator" as a more generic default password... but this IS a ublue org action... 

Should it default to "ublue-os" for the sanity of defaulting to what we've put in all our docs?

(More a question for the team)

noelmiller replied:
It would save effort in our actions as I wouldn't have to specify it every time.

bsherman replied:
@KyleGospo do you agree `ublue-os` is a better default?

If so, lets change that back as the default.  Please forgive me for causing confusion. And I'll approve then. :-)

p5 replied:
The rest of the params are defaulting to ublue values, so I vote to keep ublue-os by default.

bsherman replied:
@noelmiller go for it!

noelmiller committed:
fix: set password to ublue default

noelmiller committed:
fix: changed enrollment password in containerfile

bsherman reviewed (approved):
I think it looks good!
